core/DataAccess/Model.php

```diff
@@ -619,7 +619,7 @@ public function getArchiveIdsForSegments($archiveTableName, array $segments, $ol
     private function getDeletedSegmentWhereClause(array $segment)
     {
         $idSite = (int)$segment['enable_only_idsite'];
-        $segmentHash = Segment::getSegmentHash(urlencode($segment['definition']));
+        $segmentHash = Segment::getSegmentHash($segment['definition']);
```

Comment:
Here, maybe we should use the hash in the DB rather than computing from the definition?

Reply:
@diosmosis do you know why that would be needed here? The logic should maybe still work here too?

Reply:
The definition is still encoded differently than the `segment` parameter? Or has how we need to encode them changed? Either way, wouldn't it be safer to use the hash in the DB (which should just be `$segment['hash']`)?

Reply:
Good point, here could probably simply use `$segment['hash']` directly 👍  the method should have otherwise tried the different decode,encode etc find the right hash too (I think)

Reply:
make sense, I'll do it